utils/errcheck.go

```diff
@@ -49,90 +49,105 @@ var (
-	}
-	if errToCheck.Code == 40000 && len(errToCheck.Errors) >= 1 && errToCheck.Errors[0].Code == 40002 &&
-		strings.Contains(errToCheck.Errors[0].Message, "Email is not valid") {
+	if isAuthFailed(errToCheck) {
 		return "", ErrAuthFailed
 	}
-	if errToCheck.Code == 40000 && len(errToCheck.Errors) >= 1 && errToCheck.Errors[0].Code == 40002 &&
-		strings.Contains(errToCheck.Errors[0].Message, "Email is empty") {
-		return "", ErrAuthFailed
-	}
-	if errToCheck.Code == 40000 && len(errToCheck.Errors) >= 1 && errToCheck.Errors[0].Code == 40003 &&
-		strings.Contains(errToCheck.Errors[0].Message, "Password is invalid") {
-		return "", ErrAuthFailed
-	}
-	if errToCheck.Code == 40000 && len(errToCheck.Errors) >= 1 && errToCheck.Errors[0].Code == 40004 &&
-		strings.Contains(errToCheck.Errors[0].Message, "Password is invalid") {
+
+	switch errToCheck.Code {
+	case 40000:
+		return "", checkCode40000(errToCheck, vcli.Address)
+	case 40020:
 		return "", ErrPasswordTooWeak
-	}
-	if errToCheck.Code == 40000 && len(errToCheck.Errors) >= 1 && errToCheck.Errors[0].Code == 40003 &&
-		strings.Contains(errToCheck.Errors[0].Message, "Password is empty") {
-		return "", ErrAuthFailed
-	}
-	if errToCheck.Code == 40000 && len(errToCheck.Errors) >= 1 && errToCheck.Errors[0].Code == 40003 &&
-		strings.Contains(errToCheck.Errors[0].Message, "Access Key already registered with given name") {
-		return "", ErrApiKeyAlreadyRegistered
-	}
-	if errToCheck.Code == 40029 {
+	case 40024: // user account is already activated
+		return "", nil
+	case 40026:
+		return "", ErrInvalidConfirmationCode
+	case 40029:
 		return "", ErrEmptyMFACode
-	}
-	if errToCheck.Code == 40020 {
-		return "", ErrPasswordTooWeak
-	}
-	if errToCheck.Code == 40033 {
+	case 40033:
 		return "", ErrEmailIsNotConfirmed
+	case 40052:
+		return "", ErrEmailAlreadyRegistered
+	case 20303:
+		return "", ErrIncorrectAppToken
+	case 20308:
+		return "", ErrIncorrectAppToken
 	}
-	if errToCheck.Code == 40027 || errToCheck.Code == 40019  {
-		return "", ErrAuthFailed
+
+	fmt.Println("error sending request to " + vcli.Address)
+	return "", errToCheck
+}
+
+func isUnauthorized(err *client.VirgilAPIError) bool {
+	return err.StatusCode == http.StatusUnauthorized ||
+		err.Code == 40100 ||
+		err.Code == 20311
+}
+
+func isEntityNotFound(err *client.VirgilAPIError) bool {
+	result := err.StatusCode == http.StatusNotFound ||
+		err.Code == 40015 ||
+		strings.Contains(err.Error(), "Entity was not found")
+
+	if len(err.Errors) != 0 {
+		result = result || err.Errors[0].Code == 40400
 	}
-	if errToCheck.Code == 20303 || errToCheck.Code == 20308 {
-		return "", ErrIncorrectAppToken
+	return result
+}
+
+func isAuthFailed(err *client.VirgilAPIError) bool {
+	result := err.Code == 40019 || err.Code == 40027
+	if len(err.Errors) != 0 {
+		errField := err.Errors[0]
+
+		result = result || errField.Code == 40001 && strings.Contains(errField.Message, "Invalid email or password")
```

Comment:
why check for message if you have the code?

Reply:
That code was before. I can suppose that we use the same code but different messages.